### ISSUE TITLE
Accessing Participant's Permissions

### ISSUE THREAD

**Issue by iankoex** [livekit/client-sdk-swift#119]
The variables such as `canPublish` is `private`.
In the app that I am working on I want to be able to list everyone that can publish as speakers and the rest as listeners. 
Additionally can have a field for `isAdmin` and allow for updating permissions (therefore creating moderation)

**Comment by hiroshihorie** [comment]
Wow sorry I missed this one, it's now exposed in this commit: https://github.com/livekit/client-sdk-swift/commit/f05dc6940e07dd2d92eeee2efa8d1251e4405571

It will be available in a new release. (or you can use `main` directly for now)

Can you describe the `isAdmin` field ? You can use the metadata APIs to manage your app-specific logic.

I will close this for now.

**Comment by iankoex** [comment]
Thanks for the commit. 
The participant's permission is enough to create what I intended. About the room Admins I have used the room's metadata to notify all the participants who the admins are. 
The issue that I've encountered is that `ObservableRoom` is a subclass of `RoomDelegate` but does not have the method ` func room(_ room: Room, didUpdate metadata: String?) {}` and therefore we cannot override it in our own `ObservableObject`. 
I can open a new issue for this.
